eggo/dag.py

```diff
@@ -69,6 +69,40 @@ class ToastConfig(Config):
     config = JsonFileParameter()
 
 
+def _cghub_download(url, tmp_dir, cghub_key=None, n_threads=8):
+    """Download from CGHub to TMP_DIR.
+
+    Requires GeneTorrent. Download client `gtdownload` must be
+    on PATH. Use public key if none provided and CGHUB_KEY
+    environment variable not set. Returns analysis subdirectory.
+    """
+    # 1. Check env for CGHub key and substitute public if necessary
+    if cghub_key is None:
+        cghub_key = os.environ.get('CGHUB_KEY') or CGHUB_PUBLIC_KEY
+
+    # 2. Parse url for analysis ID and filename
+    analysis_id, filename = url.lstrip('cghub://').split('/')
+
+    # 3. Download with gtdownload
+    cmd = 'gtdownload -c {keypath} -p {prefix} --max-children {threads} -v {analysis_id}'
+    p = Popen(cmd.format(keypath=cghub_key, prefix=tmp_dir, threads=n_threads,
+                         analysis_id=analysis_id), shell=True)
+    p.wait()
+
+    return os.path.join(tmp_dir, analysis_id)
```

Comment:
sure. this is to prevent nested folders in the eggo bucket, since gtdownload creates a folder named by analysis_id and stores the files inside it. I'll have it copy them out into tmp_dir instead